.github/workflows/build.yml

```diff
@@ -2,66 +2,106 @@ name: CI
+      ref: ${{ steps.check_release.outputs.ref }}
+    steps:
+      - name: Fetch Version
+        id: tag
+        run: |
+          if [[ "${{ github.event.inputs.version }}" != "" ]]; then
+            echo "version=${{ github.event.inputs.version }}" >> $GITHUB_OUTPUT
+          fi
+      - name: Update Tag
+        uses: richardsimko/update-tag@v1
+        if: ${{ steps.tag.outputs.version }}
+        with:
+          tag_name: ${{ steps.tag.outputs.version }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Check Release
+        id: check_release
+        run: |
+          export REF=`git describe --match v[0-9]* HEAD --tags`
+          echo "Build Version: $REF"
+          [[ "$REF" != "" ]] && echo "ref=$REF" >> $GITHUB_OUTPUT
+          if [[ `git describe --exact-match --match v[0-9]* HEAD --tags` ]]; then
+            echo "release=$REF" >> $GITHUB_OUTPUT
+            echo "Release Version: $REF"
+          fi
   build_linux:
     name: Linux
-    runs-on: ubuntu-22.04
+    needs: [version]
+    runs-on: ubuntu-latest
+    container: ghcr.io/lite-xl/lite-xl-build-box:latest
     strategy:
       matrix:
         config:
         - { name: "GCC", cc: gcc, cxx: g++ }
-        - { name: "clang", cc: clang, cxx: clang++ }
+        # - { name: "clang", cc: clang, cxx: clang++ }
     env:
       CC: ${{ matrix.config.cc }}
       CXX: ${{ matrix.config.cxx }}
+      ARCH: x86_64
     steps:
+      - name: System Information
+        run: |
+          bash --version
+          $CC -v
       - name: Set Environment Variables
-        if: ${{ matrix.config.cc == 'gcc' }}
         run: |
           echo "$HOME/.local/bin" >> "$GITHUB_PATH"
-          echo "INSTALL_REF=${GITHUB_REF##*/}" >> "$GITHUB_ENV"
-          echo "INSTALL_NAME=lite-xl-${GITHUB_REF##*/}-linux-$(uname -m)-portable" >> "$GITHUB_ENV"
+          echo "INSTALL_NAME=lite-xl-${{ needs.version.outputs.ref }}-$ARCH-linux" >> "$GITHUB_ENV"
       - uses: actions/checkout@v3
-      - name: Python Setup
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.9
-      - name: Update Packages
-        run: sudo apt-get update
-      - name: Install Dependencies
-        run: bash scripts/install-dependencies.sh --debug
       - name: Build
-        run: |
-          bash --version
-          bash scripts/build.sh --debug --forcefallback --portable
+        run: scripts/build.sh --forcefallback --portable --addons --debug
       - name: Package
-        if: ${{ matrix.config.cc == 'gcc' }}
-        run: bash scripts/package.sh --version ${INSTALL_REF} --debug --binary
+        run: |
+          cd build-$ARCH-linux && tar -cvf ../$INSTALL_NAME.tar lite-xl && cd .. && gzip $INSTALL_NAME.tar
+          scripts/package-appimage.sh --debug --version ${{ needs.version.outputs.ref }}
```

Comment:
Could be; any particular reason? This is, after all, "packaging". And it's one line for each.

Reply:
Mostly because this is packaging two different things.
But yeah, not really important, also considering that it's done the same way for the other platforms.

Reply:
OK, well, I'm happy to split it. It's easy enough.

Reply:
Eh, don't worry about it, it's fine like this.

Reply:
Lol, too late.